### PR TITLE
change subscription endpoint

### PR DESCRIPTION
**Pull request by magicjohnson**
* move /subscriptions to /messages/subscriptions/by_id
* change topic for subscription by id
* add verification of intent
* tests refactoring

### CODE REVIEW

**Review by koriaf — APPROVED** [review]

File: api/tests/conftest.py [hunk 1]
```diff
@@ -7,12 +8,12 @@
 from api.models import Message
 
 
-@pytest.yield_fixture(scope='session')
+@pytest.fixture(scope='session')
```
Comment (adamkerz) [inline api/tests/conftest.py:11]: 👍

File: .gitignore [hunk 2]
```diff
@@ -33,5 +33,5 @@ htmlcov/
 # test results
 /api/tests/results.xml
 
-/docker/var/minio-data
+/**/var/minio-data
```
Comment (adamkerz) [inline .gitignore:36]: Do you have a separate dev process that put files in `{repo}/var` still? I want to do what I can to support all people's dev environments, without locking us into one particular way, so keen to hear about things that are painful for you.

File: api/tests/test_views.py [hunk 3]
```diff
@@ -128,57 +130,102 @@ def test_put__when_missing_message__should_return_404(self):
...
             'hub.mode': 'subscribe',
             'hub.callback': 'https://callback.url/1',
-            'hub.topic': 'aa.bb.cc',
+            'hub.topic': 'id',
         }
-        response = self.client.post(
-            url_for('views.subscriptions'),
-            mimetype='application/x-www-form-urlencoded',
-            data=urlencode(params)
+        self.mocked_responses.add(
```
Comment (adamkerz) [inline api/tests/test_views.py:151]: 👍

File: api/views.py [hunk 4]
```diff
@@ -175,14 +186,35 @@ def _unsubscribe(self, callback, topic):
...
+
+class SubscriptionByJurisdiction(BaseSubscriptionsView):
     def get_topic(self, form_data):
         return "jurisdiction.%s" % form_data['topic']
 
 
-blueprint.add_url_rule('/subscriptions', view_func=SubscriptionsView.as_view('subscriptions'))
 blueprint.add_url_rule(
-    '/subscriptions/by_jurisdiction',
+    '/messages/subscriptions/by_jurisdiction',
```
Comment (adamkerz) [inline api/views.py:214]: 👍  I missed that, thanks!

File: .gitignore (hunk 2 above)
Comment (magicjohnson) [reply .gitignore:36]: I think it was old docker instance... so this change isn't really necessary

**Pull request merged**